src/net/direct.c

```diff
@@ -43,6 +42,8 @@ static void direct_conn_send_internal(direct_conn dc)
        tcp_write or tcp_output, this will need to be revised to avoid
        deadlock. */
     spin_lock(&dc->send_lock);
+    if (q)
+        list_insert_before(&dc->sendq_head, &q->l);
     while ((next = list_get_next(&dc->sendq_head))) {
         qbuf q = struct_from_list(next, qbuf, l);
```

Comment:
It doesn't matter really, but shadow assignment not really needed.

Reply:
Sorry, I'm not sure what you mean here...that q is defined again within this scope? I guess I didn't realize I was reusing that name for the arg...but it shouldn't hurt.